### PR TITLE
Create a new paddle build option to build documentation library-0.15.0

### PR DESCRIPTION
**Pull request by shanyi15**
test=release/0.15.0

### CODE REVIEW

**Comment by shanyi15** [comment]
the same as https://github.com/PaddlePaddle/Paddle/pull/13063 , please check.@jetfuel

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -431,24 +432,60 @@ EOF
...
+#    else
+#        # Early exit, this branch doesn't require documentation build
+#        return 0;
+#    fi
+#     # Fetch the paddlepaddle.org deploy_docs.sh from the appopriate branch
+#    export DEPLOY_DOCS_SH=https://raw.githubusercontent.com/PaddlePaddle/PaddlePaddle.org/$PPO_SCRIPT_BRANCH/scripts/deploy/deploy_docs.sh
+#    export PYTHONPATH=$PYTHONPATH:${PADDLE_ROOT}/build/python:/paddle/build/python
+#    cd ..
+#    curl $DEPLOY_DOCS_SH | bash -s $CONTENT_DEC_PASSWD $TRAVIS_BRANCH ${PADDLE_ROOT} ${PADDLE_ROOT}/build/doc/ ${PPO_SCRIPT_BRANCH}
+#    cd -
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 435-452行不需要的话，请直接去掉。

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -145,8 +146,8 @@ EOF
         -DWITH_FLUID_ONLY=${WITH_FLUID_ONLY:-OFF} \
         -DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
         -DWITH_CONTRIB=${WITH_CONTRIB:-ON} \
+        -DWITH_INFERENCE=${WITH_INFERENCE:-ON} \
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 119行和149行请保持和原来一样

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -569,7 +606,7 @@ EOF
 }
 
 function test_fluid_inference_lib() {
-    if [ ${WITH_C_API:-OFF} == "OFF" ] ; then
+    if [[ ${WITH_C_API:-OFF} == "OFF" && ${WITH_INFERENCE:-ON} == "ON" ]] ; then
```
Comment (luotao1) [inline paddle/scripts/paddle_build.sh]: 594行和609行请保持和原来一样

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**